File: C_C++/dataStructures/SierpinskiTriangle.cpp
#include<stdio.h>
int main(){
	long long int data[501]={0};
	long long int  sum[501]={0};
	data[1]=1;
	sum[1]=1;
	for(int i=2,j=1;i<501;i++){
		data[i]=data[i-1]+j*4;
		sum[i]=data[i]+sum[i-1];
		j*=3;
	}
	int n;
	while(scanf("%d",&n)!=-1){
		printf("%d\n%d\n",data[n],sum[n]);
			
	}
	return 0;
}

File: Python/SierpinskiTriangle.py
data=[0]*501
sum=[0]*501
data[1]=1
sum[1]=1
i=2 
thr=1
while(i<501):
    data[i]=data[i-1]+thr*4
    sum[i]=sum[i-1]+data[i]
    thr*=3
    i+=1
try:
    while True:
        n=input()
        n=int(n)
        print(data[n])
        print(sum[n])

except EOFError:
    pass